src/main/haskell/kore/src/Kore/Unification/Procedure.hs

```diff
@@ -0,0 +1,107 @@
+{-|
+Module      : Kore.Unification.Procedure
+Description : Unification procedure.
+Copyright   : (c) Runtime Verification, 2018
+License     : NCSA
+Maintainer  : vladimir.ciobanu@runtimeverification.com
+Stability   : experimental
+Portability : portable
+-}
+module Kore.Unification.Procedure
+    ( unificationProcedure
+    ) where
+
+import           Control.Monad.Counter
+                 ( Counter )
+import           Control.Monad.Except
+                 ( ExceptT(..)  )
+import           Data.Functor.Foldable
+import           Data.Reflection
+                 ( give )
+
+import           Kore.AST.Common
+                 ( SortedVariable )
+import           Kore.AST.MetaOrObject
+                 ( Meta, MetaOrObject, Object )
+import           Kore.AST.MLPatterns
+                 ( getPatternResultSort )
+import           Kore.AST.PureML
+                 ( PureMLPattern )
+import qualified Kore.IndexedModule.MetadataTools as MetadataTools
+                 ( MetadataTools (..) )
+import           Kore.IndexedModule.MetadataTools
+                 ( MetadataTools )
+import           Kore.Predicate.Predicate
+                 ( makeAndPredicate, makeFalsePredicate )
+import           Kore.Step.Simplification.AndTerms
+                 ( termUnification )
+import           Kore.Step.Simplification.Ceil
```

Comment:
Not mandatory, but please consider importing this qualified, it makes the code easier to follow (i.e. one can tell what makeEvaluateTerm is doing without checking the imports).